src/main/java/io/confluent/connect/jdbc/util/CachedConnectionProvider.java

```diff
@@ -106,6 +106,7 @@ public synchronized void close() {
     if (connection != null) {
       try {
         log.info("Closing connection #{} to {}", count, provider);
+        connection.rollback();
```

Comment:
Are we sure this is safe to do during regular shutdown flow where we explicitly call `cachedConnectionProvider.close()` ? In this case, there may be no transaction in flight. The javadoc  for rollback says:
```
If you are trying to execute one or more statements in a transaction and get a SQLException, call the method rollback to end the transaction and start the transaction all over again
```
It does not state what happens when you call it outside of a transaction.

Also, it seems like we would leave the connection open if the rollback threw an exception.

Reply:
These are good points; I've addressed your concerns.